PSR-4 / xPDO 3.x Model Refactor

opengeek opened the pull request:
### What does it do?
Work in progress on xPDO 3.x model refactor of the Revo core featuring PSR-4 autoloading.

### Why is it needed?
To allow the core model to be maintained via xPDO 3.x and to enable PSR-4 autoloading.

### Related issue(s)/PR(s)
n/a

- [x] Finish converting core/model/modx/processors/ to core/src/Processors/
- [ ] Convert sqlsrv model classes
- [x] Remove core/model/modx/*
- [x] Figure out what to do with manager/controllers to enable PSR-4 autoloading without losing ability to rename or relocate the manager/ directory
- [ ] Work on recommended practices for extras to add their component src to the autoloader dynamically (see xPDO->addPackage() PSR-4 support)
- [x] Refactor unit tests for the new classes

opengeek committed:
Work in progress

opengeek commented:
There is a lot still to do on this…

OptimusCrime commented:
This looks great. I was just wondering if GitHub wrongfully reports new files instead of moved files? Are all these classes brand new, or are they just moved from their previous location in core?

opengeek commented:
They are being manually converted to their new location in `core/src/` and then `core/model/modx/` will only contain a few files to serve as faux class loaders for BC purposes.

gpsietzema commented:
Nice work! Am I correct in saying that this will just require a lot of time and most strategic thinking has been done? More people can now contribute towards this PR.

Commit:
Converted search processor

Commit:
Remove unused actions staff

JoshuaLuckers commented:
> Nice work! Am I correct in saying that this will just require a lot of time and most strategic thinking has been done? More people can now contribute towards this PR.

Yes, people are welcome to help convert the processors 👍 ~(without reformatting the code)~.

alroniks commented:
Added search processor refactoring https://github.com/opengeek/revolution/pull/4

alroniks commented:
@JoshuaLuckers New processors are new files so I guess code reformatting also can be applied.

JoshuaLuckers commented:
> Figure out what to do with manager/controllers to enable PSR-4 autoloading without losing ability to rename or relocate the manager/ directory

Can't we simply move `manager/controllers` to `core/src/Manager`?  Various functions in  modManagerController have to be changed to use the autoloader for the core classes of course.

opengeek commented:
@JoshuaLuckers I don't think they should be in the core at all, honestly. Perhaps we should just leave them as is for now. We can address them at least in a separate PR if the consensus is to convert them and add them to the autoloader dynamically based on the location of the manager folder. This also needs to be able to load theme specific controllers.

opengeek commented:
I believe @Alroniks is working on the security processors.

Commit:
Security/Roles processors converted

Commit:
Security/Message processors converted

Commit:
Security/Profile processors converted

Commit:
Security/Base processors converted

Commit:
Read off usage of undeclared variables

Commit:
Security/ResourceGroup processors converted

opengeek committed:
Finish converting Processors/Element/TemplateVar/Renders/

Commit:
Security/Group/User processors converted

Commit:
Security/Group/Setting processors converted

Commit:
Security/Group/Base processors converted

Commit:
Read off usage outdated file based processor

Commit:
Security/Access/UserGroup/Source processors converted

Commit:
Security/Access/UserGroup/Context processors converted

Commit:
Security/Access/UserGroup/ResourceGroup processors converted

Commit:
Security/Access/UserGroup/Category processors converted

Commit:
Security/Access/Base processors converted

Commit:
Security/Access/Permission processors converted

Commit:
Security/Access/UserGroup/AccessNamespace processors converted

JoshuaLuckers committed:
Made a start with converting the resource processors

Commit:
Security/Access/Policy/Template processors converted

Commit:
Security/Access/Policy processors converted

JoshuaLuckers committed:
Made a start with converting the resource processors

JoshuaLuckers committed:
Made a start with converting the resource processors

JoshuaLuckers committed:
Made a start with converting the resource processors

JoshuaLuckers committed:
Using the right class name resolution

Commit:
Security/User/Group processors converted

Commit:
Security/User/Base processors converted

Commit:
Added missed system/settings processors

Commit:
Security/User/Setting processors converted

Commit:
Use proper classes for user group settings

JoshuaLuckers committed:
Forwardport of modxcms/revolution/pr/14423

rthrash commented:
This pull request has been mentioned on **Community**. There might be relevant details there:

https://community.modx.com/t/mab-2019-meeting-2-notes-may-2-2019/731/1

Commit:
Security/Forms/Set processors converted

Commit:
Security/Forms/Profile processors converted

Commit:
Removed the mention of a non-existent processor

Commit:
Converted Workspace/Theme

Commit:
Converted Workspace/Lexicon

Commit:
Converted Workspace/PackageNamespace

Commit:
Converted Workspace/Providers

Commit:
Converted Workspace/Packages

JoshuaLuckers committed:
Forwardport of modxcms/revolution/pr/14523

Commit:
Converted Source/Base

Commit:
Converted Source/Type

Commit:
Converted System/ActiveResource

Commit:
Converted System/Charset

Commit:
Converted System/ClassMap

Commit:
Converted System/ContentType

Commit:
Converted System/Country

Commit:
Converted System/ErrorLog

Commit:
Converted System/Log

Commit:
Converted System/Registry

Commit:
Converted System/Menu

Commit:
Converted System/Language

Commit:
Converted System/Rte

Commit:
Converted System/Event

Commit:
Converted System/Import

Commit:
Converted System/Derivatives

Commit:
Converted System/Base also adopting existing old code for easier further migration

Commit:
Converted System/Dashboard

Commit:
Converted System/DatabaseTable

Commit:
Remove unused Actions combo

Commit:
Added missed settings processors

Commit:
Backporting pr-14511

opengeek committed:
Add User/Create, User/Update and User/Validation processor classes

opengeek committed:
Remove old model and convert gateway files and build scripts to use new classes

Commit:
Fix error during uploading files to Media Source root

opengeek committed:
Attempt to convert unit tests

sergant210 commented:
> Figure out what to do with manager/controllers to enable PSR-4 autoloading without losing ability to rename or relocate the manager/ directory

Connectors use core classes. But what to do with the core directory? That is the question.
```
"autoload": {
        "psr-4": {
            "MODX\\Revolution\\": "core/src/"
        },
        "classmap": [
            "core/xpdo/"
        ]
},
```

opengeek committed:
Get setup working for new install mode

opengeek committed:
Reverting inadvertent changes to setup/includes/config.core.php

opengeek committed:
Fix the _build/test/properties.sample.inc.php

opengeek committed:
Fix reference to modWorkspace

opengeek committed:
More refactoring

opengeek committed:
Moved src files to subdirectory to allow specifying a physical location for package metadata

opengeek committed:
Add support for loading legacy core classes

opengeek committed:
Fix invalid declaration of modMediaSource model

Was extending modAccessibleObject instead of modAccessibleSimpleObject due to an error that has existed in the codebase since 2011

opengeek committed:
Continuing to get unit tests working...

opengeek committed:
Fix more tests

opengeek committed:
Finish unit test fixes

opengeek committed:
Fix call to legacy processor in Welcome controller

opengeek committed:
Get past login and loading manager

opengeek committed:
Initial manager view functional

opengeek committed:
Get resource edit view functional

Commit:
Get elements edit view functional

Commit:
Get system controllers functional

opengeek committed:
Merge pull request #23 from Alroniks/elements-work

Get elements edit view functional

opengeek committed:
Merge pull request #24 from Alroniks/system-controllers

Get system controllers functional

JoshuaLuckers committed:
Renamed the "workspace/namespace" actions to fix issues after the renamed processors

JoshuaLuckers committed:
Load the modManagerController to fix errors in the manager when viewing workspaces for "namespaces"

JoshuaLuckers commented:
Searching in the System Settings is broken. Those issue can be fixed after we merge this PR.

JoshuaLuckers commented:
Changing values for settings doesn't seem to work either.

JoshuaLuckers committed:
Import modManagerController

JoshuaLuckers committed:
Import undefined classes

opengeek committed:
Merge pull request #25 from JoshuaLuckers/psr-4-namespaces-bugfixes

Psr 4 namespace "workspace" bugfixes

opengeek committed:
Merge remote-tracking branch 'JoshuaLuckers/psr-4-undefined-classes' into psr4

* JoshuaLuckers/psr-4-undefined-classes:
  Import undefined classes
  Import modManagerController

opengeek committed:
Replace more legacy class references

opengeek committed:
Fix some actions and a stupid boolean bug in System/Settings/Update

opengeek committed:
Add missing platform-specific methods and fix others

opengeek committed:
Remove old upgrade scripts and only allow upgrades from 2.6+

opengeek committed:
Regenerate model with properly escaped default values

alroniks commented on setup/lang/en/default.inc.php:
```diff
@@ -213,6 +213,7 @@
 $_lang['toggle'] = 'Toggle';
 $_lang['toggle_success'] = 'Toggle Success Messages';
 $_lang['toggle_warnings'] = 'Toggle Warnings';
+$_lang['upgrade_version_unsupported'] = 'Upgrading from MODX [[+version]] is not supported by this release. You will need to upgrade to MODX 2.6 or later before upgrading to this release.';
```
Why 2.6 and not 2.7?

opengeek replied:
Probably could have supported older, but didn't want to deal with it. There were no upgrade scripts for 2.6, so no reason not to support 2.6+.

opengeek committed:
Fix default database values containing backslashes (class defs)

opengeek committed:
Fix only variables can be passed by reference error in modHashing

opengeek committed:
Begin work on making upgrades from 2.x functional

opengeek committed:
Fix schema-defined criteria missing from model maps

opengeek committed:
Convert sqlsrv model
